controllers/route_controller.go

```diff
@@ -0,0 +1,394 @@
+	gwEventHandler := eventhandlers.NewEnqueueRequestGatewayEvent(mgrClient)
+	svcEventHandler := eventhandlers.NewEqueueHTTPRequestServiceEvent(mgrClient)
+	svcImportEventHandler := eventhandlers.NewEqueueRequestServiceImportEvent(mgrClient)
+
+	routeInfos := []struct {
+		routeType      RouteType
+		gatewayApiType client.Object
+	}{
+		{HTTP, &gateway_api_v1beta1.HTTPRoute{}},
+		{GRPC, &gateway_api_v1alpha2.GRPCRoute{}},
+	}
+
+	for _, routeInfo := range routeInfos {
+		reconciler := RouteReconciler{
+			routeType:        routeInfo.routeType,
+			log:              log,
+			client:           mgrClient,
+			scheme:           mgr.GetScheme(),
+			finalizerManager: finalizerManager,
+			eventRecorder:    mgr.GetEventRecorderFor(string(routeInfo.routeType) + "route"),
+			latticeDataStore: datastore,
+			modelBuilder:     gateway.NewLatticeServiceBuilder(mgrClient, datastore, cloud),
+			stackDeployer:    deploy.NewLatticeServiceStackDeploy(cloud, mgrClient, datastore),
+			stackMarshaller:  deploy.NewDefaultStackMarshaller(),
+		}
+
+		err := ctrl.NewControllerManagedBy(mgr).
+			For(routeInfo.gatewayApiType).
+			Watches(&source.Kind{Type: &gateway_api_v1beta1.Gateway{}}, gwEventHandler).
+			Watches(&source.Kind{Type: &corev1.Service{}}, svcEventHandler).
+			Watches(&source.Kind{Type: &mcs_api.ServiceImport{}}, svcImportEventHandler).
+			Complete(&reconciler)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=grpcroutes;httproutes,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=grpcroutes/status;httproutes/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=grpcroutes/finalizers;httproutes/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the Route object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
+func (r *RouteReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	return lattice_runtime.HandleReconcileError(r.reconcile(ctx, req))
+}
+
+func (r *RouteReconciler) reconcile(ctx context.Context, req ctrl.Request) error {
+	route, err := r.getRoute(ctx, req)
+	if err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	if err := r.client.Get(ctx, req.NamespacedName, route.K8sObject()); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	if !r.isRouteRelevant(ctx, route) {
+		// not relevant
+		return nil
+	}
+
+	if !route.DeletionTimestamp().IsZero() {
+		r.log.Info("Deleting")
```

Comment:
for consistency with other controllers, r.log.Infow("reconcile, deleting", "name", req.Name)  
and log at the end that request reconciled successfully

Reply:
This seems like a more usable log. Made the change in following commit!